### PR TITLE
Added onRetryUploadForMediaClicked interface for GB

### PR DESCRIPTION
**Pull request by mzorz**
Introduces `Retry/Retry all/Remove` for failed media uploads in Gutenberg

gutenberg-mobile PR https://github.com/wordpress-mobile/gutenberg-mobile/pull/528
gutenberg PR https://github.com/WordPress/gutenberg/pull/13615

#### To test:
CASE A RETRY (single)
1. start a new draft
2. insert image block
3. choose DEVICE LIBRARY
4. choose an image (better a big one or throttle your connection to make it last)
5. once it starts uploading, turn airplane mode ON
6. observe it errors and displays `Failed to insert media.\nPlease tap for options.`
7. turn airplane mode OFF
8. tap on the image
9. observe the dialog appears with options REMOVE/RETRY ALL/ RETRY
10. tap on RETRY
11. observe the image starts to upload again, and finishes successfully.

testa

CASE B RETRY (multiple)
1. start a new draft
2. insert image block
3. insert another image block
4. choose DEVICE LIBRARY on one of them
5. choose an image (better a big one or throttle your connection to make it last - I used 2G as preferred network and turned wifi off to make it worse)
6. once it starts uploading, tap DEVICE LIBRARY on the other block and do the same
7. once you see both are slowly uploading, turn airplane mode ON.
8. observe it errors and displays `Failed to insert media.\nPlease tap for options.`
9. turn airplane mode OFF
10. tap on the image
11. observe the dialog appears with options REMOVE/RETRY ALL/ RETRY
12. tap on RETRY ALL
13. observe both images start to upload again

testb2

CASE C REMOVE
1. repeat steps 1-9 of case A
2. tap on REMOVE
3. observe the block is reset as not containing any image.

testc

CASE D: cancel
1. start a draft
2. insert an image block
3. choose an image from the device
4. while it's uploading, tap on it and observe the  "Stop uploading?" dialog appears.
5. tap YES to cancel upload, or NO to dismiss the dialog.

testd

Update release notes:

- [X] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.

### CODE REVIEW

**Comment by mzorz** [comment]
Just found an issue: realized that when an image fails to upload, then we exit the Editor, then enter again, we should show it still as failed.
Going to work on that 👌

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java [hunk 1]
```diff
@@ -3231,6 +3231,27 @@ private void requestTemporaryPermissions(DragEvent dragEvent) {
         requestDragAndDropPermissions(dragEvent);
     }
 
+    @Override
+    public void onMediaRetryAllClicked(Set<String> failedMediaIds) {
+        UploadService.cancelFinalNotification(this, mPost);
+        UploadService.cancelFinalNotificationForMedia(this, mSite);
+
+        ArrayList<MediaModel> failedMedia = new ArrayList<>();
```
Comment (marecar3) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java]: Maybe we can consider failedMedias or failedMediaList ?

**Comment by marecar3** [comment]
Nice work @mzorz!
Especially for detailed test scenarios!

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java (hunk 1 above)
Comment (mzorz) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java]: 👍 addressed in ccfe884dce

**Comment by mzorz** [comment]
> realized that when an image fails to upload, then we exit the Editor, then enter again, we should show it still as failed.

This case is covered here now as well in commits https://github.com/wordpress-mobile/WordPress-Android/pull/9158/commits/4c2d92d83bdc674c32a189f82104b3d887421bca and https://github.com/wordpress-mobile/WordPress-Android/pull/9158/commits/3f6143c19ff160c46b8db72eae145ce121388ad2

**Pull request merged**